### DISCUSSION TITLE
Is it possible to override default Terragrunt values on configuration blocks?

### DISCUSSION THREAD

**Discussion by enarmontas** [gruntwork-io/knowledge-base#662] [How do I] [answered]
For example `mock_outputs_merge_strategy_with_state` has a default value `no_merge` and I need to define it for each of my dependencies.
Is it possible to change default value to `shallow` either through command line or `terragrunt.hcl` configuration file, so all instances of `dependency` block uses that strategy? 

---

<ins datetime="2023-02-08T18:21:43Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109883">Tracked in ticket #109883</a></p>
</ins>

**Comment by gitsstewart** [comment]
@enarmontas you should be able to edit the terragrunt.hcl file to meet your needs. Under inputs (at the bottom of the file) you can pass in variables that override the defaults. You can take a look at an example where someone else overrode the defaults for their use case here https://github.com/gruntwork-io/knowledge-base/discussions/620

**Reply by enarmontas** [reply]
That's not exactly what I am looking for. I need to set default values for Terragrunt block itself. 

Let's say I have this dependency block:
```
dependency "parent" {
  config_path = "../../_folder"
  mock_outputs = {
    folder = {
      id = "folder/0123456789"
    }
  }
  mock_outputs_allowed_terraform_commands = [
    "validate", "plan"
  ]
  mock_outputs_merge_strategy_with_state = "shallow"
}
```
I simply want to define this:
```
dependency "parent" {
  config_path = "../../_folder"
  mock_outputs = {
    folder = {
      id = "folder/0123456789"
    }
  }
}
```
And make sure that Terragrunt sets `mock_outputs_allowed_terraform_commands` to ` ["validate", "plan"]` and `mock_outputs_merge_strategy_with_state = "shallow"` by default.

**Comment by pras111gw** [comment, marked as answer]
Hi @enarmontas, we do have a `dependencies` aggregator block for combining multiple `dependency` blocks, but it only supports `paths`. Unfortunately, that means currently we do not support higher level abstraction that you are looking for. Please feel free to add  an enhancement request on terragrunt repo - it does seem like a useful feature to me. Thanks!
